app/javascript/widget/mixins/nextAvailabilityTime.js

```diff
@@ -0,0 +1,256 @@
+      // Here this is used to get the difference between current day and next working day
+      if (this.workingHoursEnabled) {
+        const currentDay = this.currentDay;
+        const nextDay = this.nextDayWorkingHours.day_of_week;
+        const totalDays = 6;
+        return nextDay > currentDay
+          ? nextDay - currentDay
+          : totalDays - currentDay + nextDay;
+      }
+      return null;
+    },
+    dayNameOfNextWorkingDay() {
+      if (this.workingHoursEnabled) {
+        return this.dayNames[this.nextDayWorkingHours.day_of_week];
+      }
+      return null;
+    },
+    hoursAndMinutesBackInOnline() {
+      if (this.presentHour >= this.currentDayCloseHour) {
+        return this.getHoursAndMinutesUntilNextDayOpen(
+          this.nextDayOpenHour,
+          this.nextDayOpenMinute,
+          this.currentDayCloseHour
+        );
+      }
+      return this.getHoursAndMinutesUntilNextDayOpen(
+        this.currentDayOpenHour,
+        this.currentDayOpenMinute,
+        this.currentDayCloseHour
+      );
+    },
+    exactTimeInAmPm() {
+      return `${this.timeSlot.day !== this.currentDay ? 'tomorrow' : ''} ${
+        this.timeSlot.day !== this.currentDay ? '' : `at ${this.timeSlot.from}`
+      }`;
+    },
+    hoursLeftValue() {
+      const {
+        hours: hoursLeft,
+        minutes: minutesLeft,
+      } = this.hoursAndMinutesBackInOnline;
+      const hourString = hoursLeft === 1 ? 'hour' : 'hours';
+      const roundedUpMinLeft = Math.ceil(minutesLeft / 5) * 5; // round up minutes to nearest multiple of 5
+      return `in ${hoursLeft} ${hourString}${
+        minutesLeft > 0 ? ` and ${roundedUpMinLeft} minutes` : ''
+      }`;
+    },
+    minutesLeftValue() {
+      const { minutes: minutesLeft } = this.hoursAndMinutesBackInOnline;
+      const roundedUpMinLeft = Math.ceil(minutesLeft / 5) * 5; // round up minutes to nearest multiple of 5
+      return `in ${roundedUpMinLeft} minutes`;
+    },
+    hoursAndMinutesToBack() {
+      const {
+        hours: hoursLeft,
+        minutes: minutesLeft,
+      } = this.hoursAndMinutesBackInOnline;
+      if (hoursLeft > 3) {
+        return this.exactTimeInAmPm;
+      }
+      if (hoursLeft > 0) {
+        return this.hoursLeftValue;
+      }
+      if (minutesLeft > 0) {
+        return this.minutesLeftValue;
+      }
+      return 'in some time';
+    },
+    timeLeftToBackInOnline() {
+      if (this.dayDiff > 1 && this.presentHour >= this.currentDayCloseHour) {
+        return `on ${this.dayNameOfNextWorkingDay}`;
+      }
+      if (this.hoursAndMinutesBackInOnline.hours >= 24) {
+        return `tomorrow`;
+      }
+      return this.hoursAndMinutesToBack;
+    },
+  },
+  mounted() {
+    this.setTimeSlot();
+  },
+  methods: {
+    getNextDay(day) {
+      // This code calculates the next day of the week based on the current day. If the current day is Saturday (6), then the next day will be Sunday (0).
+      return day === 6 ? 0 : day + 1;
```

Comment:
```suggestion
      return (day + 1) % 7;
```

Please add the specs as well.